Get Measures Velocities

benhills opened the pull request:
In looking at this function, I think there was a component of the velocity which was being ignored. There are some other changes in this pull request as well, but this is the only consequential one.

I am changing from this:
```
xL=abs((x_ps_beam[0])-(x_ps_beam[1]))
yL=abs((y_ps_beam[0])-(y_ps_beam[1]))
if x_ps_beam[0]-x_ps_beam[1] < 0:
    theta_rad=math.atan(xL/yL)
    v_along=vy/math.cos(theta_rad)
else:
    theta_rad=math.atan(xL/yL)
    v_along=vy/math.sin(theta_rad)
```

to this:
```
xL = x[1] - x[0]
yL = y[1] - y[0]
theta = np.arctan(yL/xL)
data['meas_v_along'] = vx*np.cos(theta) + vy*np.cos(np.pi/2.-theta)
data['meas_v_across'] = vx*np.sin(theta) - vy*np.sin(np.pi/2.-theta)
```

benhills committed:
Clean up the function for extracting Measures velocities.

benhills committed:
Change the vector rotation for Measures velocities

I think that the conversion was ignoring one of the velocity components.
I tested this updated conversion for all combinations of +/- vx and vy
terms.

RGFell committed:
add a line on the documentation and cp old functionality but commented#

RGFell commented:
@benhills I am not 100% sure about the change. Let's discussed it on slack.

benhills commented:
Looking into this. One other change that needs to be made before this is accepted is that the angle (theta) should be calculated at every point along the path since it could change a little bit in this projection (or at least check this).

RGFell commented:
I leave a comment on slack. I think we need to define what do we want to compare. (nice gif)

benhills committed:
Add loop through all beams for Measures extraction.

benhills committed:
Array of track angles instead of using the first.

In converting from Measures Velocity (x/y) to along/across we were
originally using only the first angle. However, the angle changes over
the length of the track, so calculating a full array is preferred.

benhills committed:
Merge branch 'devel' into get_measures

benhills committed:
Update the flip_meas_sign arg in plotting function
